servers/services/src/test/java/org/projectnessie/services/authz/AccessCheckerTest.java

```diff
@@ -0,0 +1,182 @@
+    assertThatCode(checker::checkAndThrow).doesNotThrowAnyException();
+  }
+
+  @Test
+  public void allChecksPass() {
+    AccessChecker checker = newAccessChecker(checks -> Collections.emptyMap());
+    assertThatCode(checker::checkAndThrow).doesNotThrowAnyException();
+  }
+
+  @Test
+  public void doubleChecks() {
+    Check checkRefLog = Check.builder(CheckType.VIEW_REFLOG).build();
+    String msgRefLog = "no, you must not";
+
+    Check checkListTagCommits =
+        Check.builder(CheckType.LIST_COMMIT_LOG).ref(TagName.of("bar")).build();
+    String msgListTagCommits = "don't look into bar";
+
+    AccessChecker checker =
+        newAccessChecker(
+            checks ->
+                ImmutableMap.of(checkRefLog, msgRefLog, checkListTagCommits, msgListTagCommits));
+    checker
+        .canViewReference(BranchName.of("foo"))
+        .canViewRefLog()
+        .canListCommitLog(TagName.of("bar"))
+        .canReadEntries(DetachedRef.INSTANCE);
+    assertThat(checker.check())
+        .containsExactly(
+            immutableEntry(checkRefLog, msgRefLog),
+            immutableEntry(checkListTagCommits, msgListTagCommits));
+
+    assertThatThrownBy(checker::checkAndThrow)
+        .isInstanceOf(AccessControlException.class)
+        .hasMessage("%s, %s", msgRefLog, msgListTagCommits);
+  }
+
+  @Test
+  public void allCheckTypes() {
+    List<Check> allChecks =
+        Arrays.stream(CheckType.values())
+            .map(
+                t -> {
+                  Builder b = Check.builder(t);
+                  if (t.isRef()) {
+                    b.ref(BranchName.of("some-branch"));
+                  }
+                  if (t.isContent()) {
+                    b.key(ContentKey.of("hello", "my", "table")).contentId("cid-foo");
+                  }
+                  return b.build();
+                })
+            .collect(Collectors.toList());
+
+    Map<Check, String> expectedErrors =
+        allChecks.stream()
+            .collect(
+                Collectors.toMap(
+                    Function.identity(),
+                    c -> "no no " + c.name(),
+                    (a, b) -> a,
+                    LinkedHashMap::new));
+
+    AccessChecker checker =
+        newAccessChecker(
+            checks -> {
+              assertThat(checks).containsAnyElementsOf(allChecks);
+              return expectedErrors;
+            });
+
+    allChecks.forEach(
+        c -> {
+          switch (c.name()) {
+            case VIEW_REFERENCE:
+              checker.canViewReference(c.ref());
+              break;
+            case CREATE_REFERENCE:
+              checker.canCreateReference(c.ref());
+              break;
+            case DELETE_REFERENCE:
+              checker.canDeleteReference(c.ref());
+              break;
+            case READ_ENTRIES:
+              checker.canReadEntries(c.ref());
+              break;
+            case ASSIGN_REFERENCE_TO_HASH:
+              E:
```

Comment:
can be removed I guess?

Reply:
Can't recall that I've intentionally added that - duh

Reply:
the cat/dog was probably typing while you were away :)